tracer/src/Datadog.Trace/DataStreamsMonitoring/Aggregation/DataStreamsAggregator.cs

```diff
@@ -18,6 +20,9 @@ namespace Datadog.Trace.DataStreamsMonitoring.Aggregation;
 /// </summary>
 internal class DataStreamsAggregator
 {
+    private readonly ReaderWriterLock _lock = new();
```

Comment:
Can you use `ReaderWriterLockSlim` instead? [docs](https://learn.microsoft.com/en-us/dotnet/fundamentals/runtime-libraries/system-threading-readerwriterlockslim)